tests/simplc/Simplc.lean

```diff
@@ -0,0 +1,352 @@
+                      -- just leave in place, will become a parameter
+                      pure ()
+              go (path ++ [i+1]) goals[i]
+      else
+        -- This should be simpler, but does not work, (the
+        -- isDefEqGuarded above fails) and I do not see why
+        if let .app f x := t then
+          if (←inferType f).isArrow then
+            withoutModifingMVarAssignment do
+              let (rhs, subgoal) ← Conv.mkConvGoalFor x
+              rhs.mvarId!.setKind .natural
+              goal.assign (← mkCongrArg f subgoal)
+              go (path ++ [2]) subgoal.mvarId!
+          -- else logInfo m!"Cannot descend into dependent {f} in {t}"
+          withoutModifingMVarAssignment do
+            let (rhs, subgoal) ← Conv.mkConvGoalFor f
+            rhs.mvarId!.setKind .natural
+            goal.assign (← mkCongrFun subgoal x)
+            go (path ++ [1]) subgoal.mvarId!
+  go [] goal
+
+def mkSimpTheorems (name : Name) : MetaM SimpTheorems := do
+  let sthms : SimpTheorems := {}
+  sthms.addConst name
+
+def mkSimpTheorem (name : Name) : MetaM SimpTheorem := do
+  let sthms ← mkSimpTheorems name
+  let sthms := sthms.pre.values ++ sthms.post.values
+  return sthms[0]!
+
+def delabInspectCmd (cp : CriticalPair) : MetaM (TSyntax `command) := do
+  `(command|simp_lc inspect $(mkIdent cp.thm1.origin.key) $(mkIdent cp.thm2.origin.key))
+
+def reportBadPairs  (cmdStx? : Option (TSyntax `command)) (act : M Unit) (stats := false) : MetaM Unit := do
+  let ((.unit, badPairs), numPairs) ← M.run act
+  if stats then
+    logInfo m!"Investigated {numPairs} critical pairs"
+  unless badPairs.isEmpty do
+    logError <| m!"Found {badPairs.size} non-confluent critical pairs:" ++
+      indentD (.joinSep ((← badPairs.mapM (·.pp)).toList) "\n")
+    if let .some cmdStx := cmdStx? then
+      let mut str : String := ""
+      for cp in badPairs do
+        let stx ← delabInspectCmd cp
+        str := str ++ "\n" ++ (← PrettyPrinter.ppCategory `command stx).pretty
+      str := str ++ "\n" ++ (← PrettyPrinter.ppCategory `command cmdStx).pretty
+      TryThis.addSuggestion cmdStx { suggestion := str, messageData? := m!"(lots of simp_lc_inspect lines)" }
+
+def checkSimpLCAll (cmdStx : TSyntax `command) (root_only : Bool) (pfixs? : Option (Array Name)) : MetaM Unit := do
+  let sthms ← getSimpTheorems
+  let thms := sthms.pre.values ++ sthms.post.values
+  let thms ← thms.filterM fun sthm => do
+    if sthms.erased.contains sthm.origin then
+      return false
+    if let .decl n _ false := sthm.origin then
+      if (← isIgnoredName n) then
+        return false
+      if let some pfixs := pfixs? then
+        return pfixs.any fun pfix =>
+          if pfix = `_root_
+          then n.components.length = 1
+          else pfix.isPrefixOf n
+      return true
+    return false
+  logInfo m!"Checking {thms.size} simp lemmas for critical pairs"
+  let filtered_sthms := thms.foldl Lean.Meta.addSimpTheoremEntry (init := {})
```

Comment:
```suggestion
  logInfo m!"Checking {thms.size} simp lemmas for critical pairs"
  let filtered_sthms := thms.foldl Lean.Meta.addSimpTheoremEntry (init := {})
```

Should this `logInfo` become a trace option?

Reply:
Maybe, I'll leave this to Kim who has the most experience actually using the tool.

In general I think there is a need for information from tactics/command that you can easily get, but doesn't show up in All Messages nor the build log. Maybe a custom hover. Until then a trace works.